internal/repo/error/repoerror.go

```diff
@@ -0,0 +1,22 @@
+package error
+
+import (
+	"errors"
+)
+
+type PartialErr struct {
+	err error
+}
+
+func Partial(err error) *PartialErr {
+	return &PartialErr{err: err}
+}
+
+func (e *PartialErr) Error() string {
+	return e.err.Error()
+}
+
+func IsPartial(err error) bool {
+	var e *PartialErr
+	return err == nil || errors.As(err, &e)
```

Comment:
Let me know if you disagree, but IMO an `IsSomething(err)` function should only ever return true if there is actually an error

Reply:
I think providing a suite of the following utils should be good:
- `IsPartial` --- err is of type PartialError
- `IsPartialOrNil` --- `err == nil || IsPartial(err)` 
- `IsTotal`/`IsComplete`/`IsNonPartial` --- `err != nil && !IsPartial(err)`

That should solve all required cases, right?